src/gate.rs

```diff
@@ -196,27 +209,125 @@ impl NaryGate {
 }
 
 pub fn and_nary(graph: &mut Graph, name: &str, inputs: usize) -> NaryGate {
-    NaryGate::new(graph, name, inputs, |before, after| {
-        let mut result = before[1].sig();
-        for state in &before[2..] {
+    NaryGate::new(graph, name, inputs, |pins| {
+        let mut result = pins[1].sig();
+        // No shortcut in case of Errors
+        for state in &pins[2..] {
             result &= state.sig();
         }
 
-        after[0] = PinState::Output(result);
+        pins[0] = PinState::Output(result);
+    })
+}
+
+pub fn or_nary(graph: &mut Graph, name: &str, inputs: usize) -> NaryGate {
+    NaryGate::new(graph, name, inputs, |pins| {
+        let mut result = pins[1].sig();
+        for state in &pins[2..] {
+            result |= state.sig();
+        }
+
+        pins[0] = PinState::Output(result);
     })
 }
 
 pub fn nand_nary(graph: &mut Graph, name: &str, inputs: usize) -> NaryGate {
-    NaryGate::new(graph, name, inputs, |before, after| {
-        let mut result = before[1].sig();
-        for state in &before[2..] {
+    NaryGate::new(graph, name, inputs, |pins| {
+        let mut result = pins[1].sig();
+        for state in &pins[2..] {
             result &= state.sig();
         }
 
-        after[0] = PinState::Output(!result);
+        pins[0] = PinState::Output(!result);
+    })
+}
+
+pub fn nor_nary(graph: &mut Graph, name: &str, inputs: usize) -> NaryGate {
+    NaryGate::new(graph, name, inputs, |pins| {
+        let mut result = pins[1].sig();
+        for state in &pins[2..] {
+            result |= state.sig();
+        }
+
+        pins[0] = PinState::Output(!result);
     })
 }
 
+pub struct Buffer(Vec<Pin>);
```

Comment:
name is confusing. `fn buffer` is unary, `struct Buffer` works on a bus.

TODO: Better orthogonality between buses and pins